src/game/client.cpp

```diff
@@ -1482,6 +1482,142 @@ namespace client
     }
     //ICOMMAND(0, sendmap, "", (), sendmap());
 
+    static void startcmd(char *map, char *mode_str, char *muts_str)
+    {
+        static const struct {const char *name; int val;} mode_names[] =
+        {
+            {"demo",       G_DEMO},
+            {"edit",       G_EDITMODE},
+            {"deathmatch", G_DEATHMATCH},
+            {"dm",         G_DEATHMATCH},
+            {"capture",    G_CAPTURE},
+            {"ctf",        G_CAPTURE},
+            {"defend",     G_DEFEND},
+            {"dac",        G_DEFEND},
+            {"bomber",     G_BOMBER},
+            {"bb",         G_BOMBER},
+            {"race",       G_RACE},
+        };
+        static const struct {const char *name; int val; int modes;} mut_names[] =
+        {
+            {"multi",      1 << G_M_MULTI,      G_ALL},
+            {"ffa",        1 << G_M_FFA,        G_ALL},
+            {"coop",       1 << G_M_COOP,       G_ALL},
+            {"insta",      1 << G_M_INSTA,      G_ALL},
+            {"instagib",   1 << G_M_INSTA,      G_ALL},
+            {"medieval",   1 << G_M_MEDIEVAL,   G_ALL},
+            {"kaboom",     1 << G_M_KABOOM,     G_ALL},
+            {"duel",       1 << G_M_DUEL,       G_ALL},
+            {"survivor",   1 << G_M_SURVIVOR,   G_ALL},
+            {"classic",    1 << G_M_CLASSIC,    G_ALL},
+            {"onslaught",  1 << G_M_ONSLAUGHT,  G_ALL},
+            {"freestyle",  1 << G_M_FREESTYLE,  G_ALL},
+            {"vampire",    1 << G_M_VAMPIRE,    G_ALL},
+            {"resize",     1 << G_M_RESIZE,     G_ALL},
+            {"hard",       1 << G_M_HARD,       G_ALL},
+            {"basic",      1 << G_M_BASIC,      G_ALL},
+            {"gladiator",  1 << G_M_GSP1,       1 << G_DEATHMATCH},
+            {"oldscool",   1 << G_M_GSP2,       1 << G_DEATHMATCH},
+            {"quick",      1 << G_M_GSP1,       1 << G_CAPTURE},
+            {"defend",     1 << G_M_GSP2,       1 << G_CAPTURE},
+            {"protect",    1 << G_M_GSP3,       1 << G_CAPTURE},
+            {"quick",      1 << G_M_GSP1,       1 << G_DEFEND},
+            {"king",       1 << G_M_GSP2,       1 << G_DEFEND},
+            {"hold",       1 << G_M_GSP1,       1 << G_BOMBER},
+            {"basket",     1 << G_M_GSP2,       1 << G_BOMBER},
+            {"attack",     1 << G_M_GSP3,       1 << G_BOMBER},
+            {"timed",      1 << G_M_GSP1,       1 << G_RACE},
+            {"endurance",  1 << G_M_GSP2,       1 << G_RACE},
+            {"gauntlet",   1 << G_M_GSP3,       1 << G_RACE},
+        };
+
+        int nextmode = G_DEATHMATCH;
+        int nextmuts = 0;
+
+        if(mode_str && isnumeric(*mode_str))
+        {
+            nextmode = atoi(mode_str);
+        }
+        else if(mode_str && *mode_str)
```

Comment:
We prefer proper pointer comparisons. Please replace with e.g.,

```suggestion
        else if(mode_str != nullptr && *mode_str != '\0')
```

or something like that.

Reply:
I usually try to follow the existing style when I write code. What are we trying to do here? Are we trying to change the style of the code? Then what's the new style that we should follow?